src/odemis/gui/cont/tabs.py

```diff
@@ -1602,6 +1611,419 @@ def get_display_priority(cls, main_data):
+
+    def _on_show_advanced(self, evt):
+        """
+        Event handler for the Advanced button to show/hide stage advanced panel
+        """
+        self.panel.pnl_advanced_align.Show(evt.isDown)
+
+    def _show_cancel_warning_msg(self, txt_warning):
+        """
+        Show warning message under progress bar, hide if no message is indicated
+        """
+        self.panel.pnl_ref_msg.Show(txt_warning is not None)
+        if txt_warning:
+            self.panel.txt_warning.SetLabel(txt_warning)
+
+    # Define angle type enumeration
+    MILL, ROTATE = 0, 1
+
+    def _milling_angle_changed(self, pos):
+        """
+        Called from the milling axis connector when the stage rx change.
+        Updates the milling control with the changed angle position.
+        :param pos: (float) value of rx
+       """
+        # Milling angle is calculated using this formula: ION_BEAM_TO_SAMPLE_ANGLE – Rx
+        milling_value = self.ion_to_sample - pos
+        milling_value = math.degrees(milling_value)
+        self.panel.ctrl_milling.Value = to_string_pretty(milling_value, unit="°")
+
+    def _milling_ctrl_changed(self):
+        """
+        Called when the milling control value is changed.
+        Used to return the correct rx angle value.
+        :return: (float or None) The calculated rx angle from the milling ctrl
+        """
+        milling_angle = self._get_tilt_angle_from_ctrl(self.MILL)
+        if milling_angle is None:
+            return
+        # Store current milling angle (in case the stage moved out of milling)
+        # TODO: Check if milling angle to be stored is not the same as imaging rx
+        self._prev_milling_angle = milling_angle
+        rx_angle = self.ion_to_sample - milling_angle
+        # Set end pos rx value (for progress indication)
+        stage = self.tab_data_model.main.stage
+        self.start_pos = stage.position.value
+        self.end_pos = copy.copy(self.start_pos)
+        self.end_pos['rx'] = rx_angle
+        return rx_angle
+
+    def _get_tilt_angle_from_ctrl(self, angle_type):
+        """
+        Get the corresponding angle value from its designated UI control
+        :param angle_type: (int) either MILL, ROTATE
+        :returns (float or None) the tilt angle value in radians
+        """
+        def get_angle_value(angle_ctrl):
+            # Read the angle value from the ctrl and convert its value to radians
+            angle_value, _, _ = decompose_si_prefix(angle_ctrl.Value, unit="°")
+            angle_value = math.radians(float(angle_value))
+            return angle_value
+
+        if angle_type == self.MILL:
+            try:
+                milling_angle = get_angle_value(self.panel.ctrl_milling)
+            except ValueError as error:
+                logging.error(error)
+                return
+            return milling_angle
+        elif angle_type == self.ROTATE:
+            raise NotImplementedError("Rotation angle control is not implemented.")
```

Comment:
Removed the code for handling RZ control but it's the same handling as the milling angle.